File: merge_code.js
const fs = require('fs');
const path = require('path');

// 합칠 파일 확장자
const EXTENSIONS = ['.ts', '.tsx', '.js', '.jsx', '.css'];
// 제외할 폴더
const IGNORE_DIRS = ['node_modules', '.next', '.git', 'dist', 'build'];
// 타겟 폴더 (소스코드 위치)
const SRC_DIR = './src';
// 결과 파일명
const OUTPUT_FILE = 'full_project_code.txt';

function getAllFiles(dirPath, arrayOfFiles) {
  const files = fs.readdirSync(dirPath);

  arrayOfFiles = arrayOfFiles || [];

  files.forEach(function(file) {
    const fullPath = path.join(dirPath, file);
    
    if (fs.statSync(fullPath).isDirectory()) {
      if (!IGNORE_DIRS.includes(file)) {
        arrayOfFiles = getAllFiles(fullPath, arrayOfFiles);
      }
    } else {
      if (EXTENSIONS.includes(path.extname(file))) {
        arrayOfFiles.push(fullPath);
      }
    }
  });

  return arrayOfFiles;
}

function mergeFiles() {
  if (!fs.existsSync(SRC_DIR)) {
    console.log(`❌ Error: '${SRC_DIR}' 폴더를 찾을 수 없습니다.`);
    return;
  }

  const allFiles = getAllFiles(SRC_DIR);
  let content = "Project: The Collegiate Grill POS System\nDescription: Full Source Code Context\n\n";

  allFiles.forEach(filePath => {
    try {
      const fileContent = fs.readFileSync(filePath, 'utf8');
      // AI가 파일 경로를 잘 알아보도록 헤더 추가
      content += `\n${'='.repeat(50)}\n`;
      content += `FILE PATH: ${filePath}\n`;
      content += `${'='.repeat(50)}\n`;
      content += fileContent + "\n";
    } catch (err) {
      console.log(`Skipping file ${filePath}: ${err.message}`);
    }
  });

  fs.writeFileSync(OUTPUT_FILE, content, 'utf8');
  console.log(`✅ 성공! 모든 코드가 '${OUTPUT_FILE}' 파일 하나로 합쳐졌습니다.`);
}

mergeFiles();

File: merge_code.py
import os

# 합치고 싶은 파일 확장자들
EXTENSIONS = ('.ts', '.tsx', '.js', '.jsx', '.css')
# 제외하고 싶은 폴더명
IGNORE_DIRS = {'node_modules', '.next', '.git', 'dist', 'build'}

def merge_files(start_path, output_filename):
    with open(output_filename, 'w', encoding='utf-8') as outfile:
        # 파일 상단에 프로젝트 요약 정보 적어주기
        outfile.write("Project: The Collegiate Grill POS System\n")
        outfile.write("Description: Full Source Code Context\n\n")

        for root, dirs, files in os.walk(start_path):
            # 제외할 폴더 걸러내기
            dirs[:] = [d for d in dirs if d not in IGNORE_DIRS]

            for file in files:
                if file.endswith(EXTENSIONS):
                    file_path = os.path.join(root, file)
                    # 상대 경로로 표시 (예: src/app/page.tsx)
                    relative_path = os.path.relpath(file_path, start_path)
                    
                    try:
                        with open(file_path, 'r', encoding='utf-8') as infile:
                            # AI가 파일 구분을 잘 하도록 헤더 추가
                            outfile.write(f"\n{'='*50}\n")
                            outfile.write(f"FILE PATH: {relative_path}\n")
                            outfile.write(f"{'='*50}\n")
                            outfile.write(infile.read())
                            outfile.write("\n")
                    except Exception as e:
                        print(f"Skipping file {file}: {e}")

    print(f"✅ 성공! 모든 코드가 '{output_filename}' 파일 하나로 합쳐졌습니다.")

if __name__ == "__main__":
    # src 폴더의 내용을 긁어옵니다
    if os.path.exists('./src'):
        merge_files('./src', 'full_project_code.txt')
    else:
        print("❌ Error: 'src' 폴더를 찾을 수 없습니다. 프로젝트 루트에서 실행해 주세요.")